tests/test_backend.py

```diff
@@ -38,3 +38,19 @@ def test_future_env_select(self):
 
         with self.assertRaises(NotImplementedError):
             PyRDF.use("dask")
+
+class BackendInitTest(unittest.TestCase):
+    """
+    Tests to ensure that the Backend abstract
+    class cannot be instantiated.
+
+    """
+    def test_backend_init_error(self):
+        """
+        Test case to check that any attempt to
+        instantiate the `Backend` abstract class
+        results in a `TypeError`.
+
+        """
+        with self.assertRaises(TypeError):
+            backend = PyRDF.backend.Backend.Backend()
```

Comment:
Small thing: you do not need to even create a variable here, just `PyRDF.backend.Backend.Backend()`.

I do not thing you need any other test. In the call this morning we mentioned a test that creates a subclass of `Backend` without the `execute` method and is expected to fail, but that would be testing Python, not really our code.

Reply:
Or after thinking about @JavierCVilla 's comment, perhaps we can just add that last test to double check the `@abstractmethod` decorator is not removed from the base class.

Reply:
I had not seen @etejedor 's review, so I removed mine. Basically, I was just proposing to test not only `Backend` but a subclass of `Backend`, which might be just a dummy class created in the same test, without defining its `execute` method.

Reply:
@JavierCVilla and @etejedor sounds good. I'll that test too. But there's a small problem before that. I just found that python 2 doesn't have `ABC` class. So, I'm finding solutions common to both

Reply:
@shravan97 I think Python 2 also has an `abc` builtin module, have a look at this [link](https://docs.python.org/2/library/abc.html). Could you try using `__metaclass__` and `ABCMeta` instead?

Reply:
@JavierCVilla , yes. I tried this. In this case only Python2 threw me an error on instantiation and Python3 didn't. But I found another fix for that here : [https://stackoverflow.com/questions/35673474/using-abc-abcmeta-in-a-way-it-is-compatible-both-with-python-2-7-and-python-3-5](https://stackoverflow.com/questions/35673474/using-abc-abcmeta-in-a-way-it-is-compatible-both-with-python-2-7-and-python-3-5)